### PR TITLE
Make marshal.MustFoo not go via defer-recover

### PR DESCRIPTION
**Pull request by kalman**

### CODE REVIEW

**Comment by kalman** [comment]
The panic-recover thing makes the stack trace harder to read.

File: go/marshal/decode.go [hunk 1]
```diff
@@ -89,8 +88,13 @@ func Unmarshal(v types.Value, out interface{}) (err error) {
 // Unmarshals a Noms value into a Go value using the same rules as Unmarshal().
 // Panics on failure.
 func MustUnmarshal(v types.Value, out interface{}) {
-	err := Unmarshal(v, out)
-	d.PanicIfError(err)
+	rv := reflect.ValueOf(out)
```
Comment (arv) [inline go/marshal/decode.go]: Can you do the refactoring for this case too to prevent code duplication?

Comment (kalman) [inline go/marshal/decode.go]: I tried that at first and it wasn't a lot more readable. Let me know what you think of it now. Much more de-duplication would make it worse IMO.

**Comment by arv** [comment]
2640880 LGTM

**Pull request merged**